service-base/src/main/java/org/eclipse/hono/service/credentials/CompleteBaseCredentialsService.java

```diff
@@ -89,20 +94,32 @@
                     HttpURLConnection.HTTP_BAD_REQUEST,
                     "missing payload"));
         } else {
-            try {
-                payload.checkValidity(this::checkSecret);
-                final Future<CredentialsResult<JsonObject>> result = Future.future();
-                add(tenantId, JsonObject.mapFrom(payload), result.completer());
-                return result.map(res -> {
-                    return request.getResponse(res.getStatus())
-                            .setDeviceId(payload.getDeviceId())
-                            .setCacheDirective(res.getCacheDirective());
-                });
-            } catch (IllegalStateException e) {
-                return Future.failedFuture(new ClientErrorException(
-                        HttpURLConnection.HTTP_BAD_REQUEST,
-                        e.getMessage()));
-            }
+            final Future<EventBusMessage> requestTracker = Future.future();
+            getVertx().executeBlocking(blockingCodeHandler -> {
+                log.debug("hashing password on vert.x worker thread [{}]", Thread.currentThread().getName());
+                hashPlainPasswords(payload);
```

Comment:
IMHO we should only run the *hashPlainPasswords* method call on the vert.x worker thread pool
that would make the code more readable and would also keep most of the processing of the request on the original event loop (as before this change)

Reply:
I tried that and it didn't look more readable to me. But I will add a commit and we can decide then.

Reply:
I added a change.